examples/kistler_5074b.rs

```diff
@@ -0,0 +1,199 @@
+                }
+                // Donzo
+                subdevice
+                    .sdo_write::<u8>(SM_ASSIGNMENT, 0, 4 * 5)
+                    .await
+                    .expect("setting PDO assignment count");
+
+                (LEN_BYTES, SM_IDX, FMMU_IDX)
+            }
+            PdoDirection::MasterWrite => {
+                const LEN_BYTES: u16 = 4;
+                const SM_IDX: u8 = 2;
+                const SM_ASSIGNMENT: u16 = SM_ASSIGNMENT_BASE_ADDRESS + (SM_IDX as u16);
+                const FMMU_IDX: usize = 0;
+
+                // PDO mapping: All PDOs are fixed, nothing to do.
+
+                // PDO assignments
+                // XXX: You have to play this little song and dance with the CoE array's length.
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 0, 0_u8)
+                    .await
+                    .expect("zeroing pdo assignment count");
+                // Control Ch - BYTE
+                // XXX: Even though these are required PDO assignments, you do have to reassign
+                // them. This may have been what was breaking EtherCrab's EEPROM based setup.
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 1, 0x1600_u16)
+                    .await
+                    .expect("setting CH1 control PDO assignment");
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 2, 0x1610_u16)
+                    .await
+                    .expect("setting CH2 control PDO assignment");
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 3, 0x1620_u16)
+                    .await
+                    .expect("setting CH3 control PDO assignment");
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 4, 0x1630_u16)
+                    .await
+                    .expect("setting CH4 control PDO assignment");
+                // Donzo
+                subdevice
+                    .sdo_write(SM_ASSIGNMENT, 0, 4_u8)
+                    .await
+                    .expect("setting PDO assignment count");
+
+                (LEN_BYTES, SM_IDX, FMMU_IDX)
+            }
+        };
+
+        let len_bits = 8 * len_bytes;
+
+        let sm_config = subdevice
+            .write_sm_config(sm_idx, &sync_managers[usize::from(sm_idx)], len_bytes)
+            .await
+            .expect("configuring sync manager");
+        let desired_sm_type = match direction {
+            PdoDirection::MasterRead => SyncManagerType::ProcessDataRead,
+            PdoDirection::MasterWrite => SyncManagerType::ProcessDataWrite,
+        };
+        subdevice
+            .write_fmmu_config(len_bits, fmmu_idx, global_offset, desired_sm_type, &sm_config)
+            .await
+            .expect("configuring fmmu");
+
+        Ok(PdiSegment {
+            bit_len: usize::from(len_bits),
+            bytes:   start_offset.up_to(*global_offset),
+        })
```

Comment:
These are the important lines that this PR facilitates; the ability for a user to configure the SMs and FMMUs of their subdevices themselves.